include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,550 @@
+
+    //!\copydoc begin()
+    auto cbegin() const noexcept
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return begin();
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the range.
+     * \returns Iterator to the end.
+     *
+     * \details
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    auto end() noexcept
+    {
+        return std::ranges::end(urange);
+    }
+
+    //!\copydoc end()
+    auto end() const noexcept
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return std::ranges::end(urange);
+    }
+
+    //!\copydoc end()
+    auto cend() const noexcept
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return end();
+    }
+    //!\}
+};
+
+template <std::ranges::view urng_t>
+template <typename rng_t>
+class minimiser<urng_t>::window_iterator
+{
+private:
+    //!\brief The iterator type of the underlying range.
+    using it_t = std::ranges::iterator_t<rng_t>;
+    //!\brief The sentinel type of the underlying range.
+    using sentinel_t = std::ranges::sentinel_t<rng_t>;
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief Type for distances between iterators.
+    using difference_type = typename std::iter_difference_t<it_t>;
+    //!\brief Value type of this iterator.
+    using value_type = size_t;
+    //!\brief The pointer type.
+    using pointer = void;
+    //!\brief Reference to `value_type`.
+    using reference = value_type;
+    //!\brief Tag this class as input iterator.
+    using iterator_category = std::input_iterator_tag;
+    //!\brief Tag this class depending on which concept `it_t` models.
+    using iterator_concept = std::conditional_t<std::contiguous_iterator<it_t>,
+                                                typename std::random_access_iterator_tag,
+                                                seqan3::iterator_tag_t<it_t>>;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr window_iterator()                                    = default; //!< Defaulted.
+    constexpr window_iterator(window_iterator const &)             = default; //!< Defaulted.
+    constexpr window_iterator(window_iterator &&)                  = default; //!< Defaulted.
+    constexpr window_iterator & operator=(window_iterator const &) = default; //!< Defaulted.
+    constexpr window_iterator & operator=(window_iterator &&)      = default; //!< Defaulted.
+    ~window_iterator()                                             = default; //!< Defaulted.
+
+    /*!\brief Construct from a given iterator of std::size_t, a second iterator of std::size_t and w_elems.
```

Comment:
Again, the description says w_elems but the parameter is named `w`. Please keep consistent naming.

Reply:
```suggestion
    /*!\brief Construct from begin and end iterators of a given range over std::size_t elements, and the number elements per window.
```

Reply:
Make it simple (details are in the param description already): 
Construct from two iterators and the number of elements per window.